### PR TITLE
Remove duplicate function prototype

### LINKED ISSUE #6520: Function declaration duplicate?
The function `mbedtls_cipher_info_from_psa` is declared in 
https://github.com/Mbed-TLS/mbedtls/blob/55fd0b9fc1a2a8e0720c8ab15e06cea579aa4f4b/library/psa_crypto_cipher.h#L39
and in 
https://github.com/Mbed-TLS/mbedtls/blob/55fd0b9fc1a2a8e0720c8ab15e06cea579aa4f4b/library/psa_crypto_core.h#L261

The second seems to be a superfluous duplicate.

Signed-off-by: Oberon microsystems

### PR DESCRIPTION
**Pull request by daverodgman**
Fixes #6520 

Changelog: not needed - this is an internal header
Backport: #6525

### CODE REVIEW

File: library/psa_crypto_cipher.h [hunk 1]
```diff
@@ -24,22 +24,6 @@
 #include <mbedtls/cipher.h>
 #include <psa/crypto.h>
 
-/** Get Mbed TLS cipher information given the cipher algorithm PSA identifier
```
Comment (gilles-peskine-arm) [inline library/psa_crypto_cipher.h:27]: This declaration belongs here. Please remove the copy in `psa_crypto_core.h` instead.

Comment (daverodgman) [reply library/psa_crypto_cipher.h:27]: It looked to me like it wasn't a good fit for this file - the docs at the top of psa_crypto_cipher.h say "PSA cipher driver entry points", and this isn't a driver entry point? Happy to switch it round if preferred though.

Comment (gilles-peskine-arm) [reply library/psa_crypto_cipher.h:27]: The function is defined in `psa_crypto_cipher.c`. The description should probably have “and associated auxiliary functions” or some such.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**